web-ui/src/pages/PulseReportPage.jsx

```diff
@@ -307,6 +355,50 @@ const PulseReportPage = () => {
     </Card>
   );
 
+  const handleCommentClick = (pulse) => {
```

Comment:
When the comment icon at the end of a row of pulse data is clicked, this function is called.
It triggers displaying the internal and external comments associated with that pulse entry in a modal dialog.